### PR TITLE
Update DG content to reflect updates better

### PR DESCRIPTION
**Pull request by AdityaB4**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -3,22 +3,30 @@ layout: page
 title: Developer Guide
 ---
 * Table of Contents
-{:toc}
+- [Acknowledgements](#acknowledgements)
+- [Setting Up](#setting-up)
+- [Design](#design)
+- [Implementation](#implementation)
+- [Misc. Docs](#misc-documentation)
+- [Appendix](#appendix)
```
Comment (yadunut) [inline docs/DeveloperGuide.md:11]: I think the {:toc} automatically generates the table of contents with jekyll, as you can see from the existing site, where the TOC is inserted. 
https://ay2324s2-cs2103t-w12-2.github.io/tp/DeveloperGuide.html, so I don't think this needs to be done.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -3,22 +3,30 @@ layout: page
...
 
 --------------------------------------------------------------------------------------------------------------------
 
+<a id="acknowledgements"></a>
```
Comment (yadunut) [inline docs/DeveloperGuide.md:15]: The Generated TOC links to eac header also, so this doesn't need to be done too.

**Review by yadunut — COMMENTED** [review]
Some comments regarding the autogeneration of ToC.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (AdityaB4) [reply docs/DeveloperGuide.md:11]: The idea here was to have finer control over nesting without having to change heading levels (`#`, `##` etc.). We can always revert this later if it ends up being something we won't use.

**Review by blaukc — APPROVED** [review]
i think yadunand is right abt the toc thing but ig this is fine as long as we rmb to update the toc now

**Pull request merged**